astropy/io/ascii/cparser.pyx

```diff
@@ -397,18 +398,21 @@ cdef class CParser:
         except (ImportError, NotImplementedError, AttributeError, OSError):
             self.raise_error("shared semaphore implementation required "
                              "but not available")
-        cdef int offset = self.tokenizer.source_pos
+        cdef off_t offset = self.tokenizer.source_pos
 
         if offset == source_len: # no data
             return (dict((name, np.array([], dtype=np.int_)) for name in
                          self.names), [])
 
-        cdef int chunksize = math.ceil((source_len - offset) / float(N))
+        cdef long chunksize = math.ceil((source_len - offset) / float(N))
         cdef list chunkindices = [offset]
 
         # This queue is used to signal processes to reconvert if necessary
         reconvert_queue = multiprocessing.Queue()
 
+        cdef int i
```

Comment:
whats the expected size of N? is it guaranteed to be smaller than 32 bit?
or just don't risk it and use large enough type